hazenlib/ghosting.py

```diff
@@ -131,75 +132,95 @@ def get_eligible_area(signal_bounding_box, dcm, slice_radius=5):
+                          np.newaxis], np.array(
+                range(upper_row, lower_row))
 
     else:
         if upper_row < dcm.Rows / 2:
             # signal is in top half
             eligible_rows = range(lower_row + padding_from_box, dcm.Rows - slice_radius)
             eligible_columns = range(left_column, right_column)
+            ghost_slice = np.array(
+            range(lower_row + padding_from_box, dcm.Rows - slice_radius), dtype = np.intp)[:,
+            np.newaxis], np.array(
+        range(left_column, right_column))
         else:
             # signal is in bottom half
             eligible_rows = range(slice_radius, upper_row - padding_from_box)
             eligible_columns = range(left_column, right_column)
+            ghost_slice = np.array(
+                range(slice_radius, upper_row - padding_from_box), dtype=np.intp)[:,
+                          np.newaxis], np.array(
+                range(left_column, right_column))
 
     return eligible_columns, eligible_rows
 
 
-def get_ghost_slice(signal_bounding_box, dcm, slice_radius=5):
-    max_mean = 0
-    max_index = (0, 0)
-    windows = {}
-    arr = dcm.pixel_array
-
-    eligible_columns, eligible_rows = get_eligible_area(signal_bounding_box, dcm, slice_radius)
-    for idx, centre_voxel in np.ndenumerate(arr):
-        if idx[0] not in eligible_rows or idx[1] not in eligible_columns:
-            continue
-        else:
-            windows[idx] = arr[idx[0]-slice_radius:idx[0]+slice_radius, idx[1]-slice_radius:idx[1]+slice_radius]
 
-    for idx, window in windows.items():
-        if np.mean(window) > max_mean:
-            max_mean = np.mean(window)
-            max_index = idx
 
+
+def get_ghost_slice(signal_bounding_box, dcm, slice_radius=5):
+    eligible_area = get_eligible_area(signal_bounding_box, dcm, slice_radius=slice_radius)
     ghost_slice = np.array(
-        range(max_index[1] - slice_radius, max_index[1] + slice_radius), dtype=np.intp)[:, np.newaxis], np.array(
-        range(max_index[0] - slice_radius, max_index[0] + slice_radius)
+        range(min(eligible_area[1]),max(eligible_area[1])), dtype=np.intp)[:, np.newaxis], np.array(
+        range(min(eligible_area[0]), max(eligible_area[0]))
     )
-    #ghost_slice returns columns, rows of indexes
     return ghost_slice
 
 
-def get_ghosting(dcm, plotting=False) -> dict:
+
+
+
+
+
+
+def get_ghosting(dcm, report_path=False) -> dict:
 
     bbox = get_signal_bounding_box(dcm.pixel_array)
 
     x, y = hazenlib.get_pixel_size(dcm)  # assume square pixels i.e. x=y
     # ROIs need to be 10mmx10mm
     slice_radius = int(10//(2*x))
 
+
     signal_centre = [(bbox[0]+bbox[1])//2, (bbox[2]+bbox[3])//2]
     background_rois = get_background_rois(dcm, signal_centre)
     ghost_col, ghost_row = get_ghost_slice(bbox, dcm, slice_radius=slice_radius)
-    ghost = dcm.pixel_array[(ghost_row, ghost_col)]
+    ghost = dcm.pixel_array[(ghost_col, ghost_row)]
```

Comment:
Is this an intentional change, swapping around ghost_col / ghost_row? Just checking in case you left if that way round during testing.

Reply:
Hi Tom, thank you for all the comments. No, this one is not intentional.